### PR TITLE
Generate unit tests for LRO discovery

### PR DESCRIPTION
**Pull request by andreamlin**

### CODE REVIEW

**Comment by andreamlin** [comment]
PTAL

File: src/main/resources/com/google/api/codegen/java/http_test.snip [hunk 1]
```diff
@@ -90,6 +90,46 @@
...
+    {@addException(test)}
+
+    try {
+      {@initCode(test.testCaseInitCode)}
+
+      client.{@test.clientMethodName}(\
+          {@sampleMethodCallArgList(test.testCaseInitCode.fieldSettings)}).get();
+      Assert.fail("No exception raised");
+    } catch (ExecutionException e) {
+      Assert.assertEquals(InvalidArgumentException.class, e.getCause().getClass());
```
Comment (vam-google) [inline src/main/resources/com/google/api/codegen/java/http_test.snip]: Maybe
```java
Assert.assertTrue(e.getCause() instanceof InvalidArgumentException);
```
is cleaner, and also allows subclasses of `InvalidArgumentException` as valid cause. This seems lika a desired behavior. In real world `catch(InvalidArgumentException e)` will also catch a subclass of `InvalidArgumentException` which is idiomatic for java exceptions, so the tests should treat `InvalidArgumentException` and its potential subclasses as equally correct.

**Comment by hzyi-google** [comment]
@vam-google Do you think this PR is still needed?

**Comment by chrisdunelm** [comment]
Old PR. Closing.

**Pull request closed without merging**